lms/djangoapps/course_blocks/transformers/hidden_content.py

```diff
@@ -66,6 +67,13 @@ def collect(cls, block_structure):
             xblock_field_name='hide_after_due',
             merged_field_name=cls.MERGED_HIDE_AFTER_DUE,
         )
+        collect_merged_date_field(
+            block_structure,
+            transformer=cls,
+            xblock_field_name='end',
+            merged_field_name=cls.MERGED_END_DATE,
+            default_date=MAXIMUM_DATE
+        )
 
         block_structure.request_xblock_fields('self_paced', 'end', 'due')
```

Comment:
@ziafazal, is there a reason we haven't [removed the `end` field](https://github.com/openedx/edx-platform/pull/30589/files#diff-ff080e1d7c7980cab4bb38b95d84406038c9bc6d77d8a05659b2c53791885678L70-R88) in this PR?

```suggestion
        block_structure.request_xblock_fields('self_paced', 'due')
```

Reply:
@Agrendalath I did not remove `end` field to keep block transformer schema backward compatible as suggested in this [doc](https://openedx.atlassian.net/wiki/spaces/AC/pages/154861855/Block+Structure+Cache+Invalidation+Proposal#BlockStructureCacheInvalidationProposal-TransformerVersionUpdates)